ethereum/nft/trades/insert_cryptopunks.sql

```diff
@@ -0,0 +1,165 @@
+CREATE OR REPLACE FUNCTION nft.insert_cryptopunks(start_ts timestamptz, end_ts timestamptz=now(), start_block numeric=0, end_block numeric=9e18) RETURNS integer
+LANGUAGE plpgsql AS $function$
+DECLARE r integer;
+BEGIN
+
+WITH rows AS (
+    INSERT INTO nft.trades (
+	block_time,
+	nft_project_name,
+	nft_token_id,
+	platform,
+	platform_version,
+	category,
+	evt_type,
+	usd_amount,
+	seller,
+	buyer,
+	original_amount,
+	original_amount_raw,
+	original_currency,
+	original_currency_contract,
+	currency_contract,
+	nft_contract_address,
+	exchange_contract_address,
+	tx_hash,
+	block_number,
+	tx_from,
+	tx_to,
+	trace_address,
+	evt_index,
+	trade_id
+    )
+
+    SELECT
+        trades.evt_block_time AS block_time,
+        'CryptoPunks' AS nft_project_name,
+        CAST(trades."punkIndex" AS TEXT) AS nft_token_id,
+        'LarvaLabs Contract' AS platform,
+        '1' AS platform_version,
+        'Buy' AS category,
+        'Trade' AS evt_type,
+        trades.value / 10 ^ 18 * p.price AS usd_amount,
+        trades."fromAddress" AS seller,
+        trades."toAddress" AS buyer,
+        trades.value / 10 ^ 18 AS original_amount,
+        trades.value AS original_amount_raw,
+        'ETH' AS original_currency,
+        '\x0000000000000000000000000000000000000000'::bytea AS original_currency_contract,
+        '\xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2'::bytea AS currency_contract,
+        trades.contract_address AS nft_contract_address,
+        trades.contract_address AS exchange_contract_address,
+        trades.evt_tx_hash AS tx_hash,
+        trades.evt_block_number AS block_number,
+        tx."from" AS tx_from,
+        tx."to" AS tx_to,
+        NULL::integer[] AS trace_address,
+        trades.evt_index,
+        row_number() OVER (PARTITION BY 4, 18, 23, 6) AS trade_id
```

Comment:
As far as I am aware. You don't need trade id because I don't think multiple trades are ever grouped in a trace or a log. Am I wrong on this? If you can remove trade id it will be good because it is a source of many bugs.

Reply:
Also can you refer to the columns by name and not index. It is very hard to review the code when one needs to count the column index etc. Also makes it very susceptible to bugs caused by updating the record set list.

Reply:
This is a really though call for me. From what i've seen, multiple trades are not grouped in a trace or log. Discussed this with @0xBoxer. Aggregators are coming in the nft space, and this seems a strong argument for keeping trade_id in there.

Reply:
I absolutely get your point about referring to the column names by name and not by index. The reason why i did not do so, is because i had bugs while testing with columns names in the web interface. The only way to get consistent behavior was to use indexes. If you insist on this change, i will happily oblige, but it is not a quick replace edit but more a matter of refactoring and testing.

Reply:
Thanks for the explanation. What bugs did you have during testing? It would help the readability and future debugging of this a lot if we used the names instead of indexes. Thank you!

Reply:
I had inconsistencies between using indexes and names, but upon further testing it turns out that using indexes does not produce accurate results.Thank you so much for insisting on the usage of names. Refactored and upcoming commit will use names.